code/modules/cargo/expressconsole.dm

```diff
@@ -162,21 +162,26 @@
 			if(!(obj_flags & EMAGGED))
 				if(SO.pack.cost <= SSshuttle.points)
 					landingzone = locate(/area/quartermaster/storage) in GLOB.sortedAreas
+					if (!landingzone)
+						landingzone = get_area(src)
 					for(var/turf/open/floor/T in landingzone.contents)//uses default landing zone
 						if(is_blocked_turf(T))
 							continue
 						LAZYADD(empty_turfs, T)
 						CHECK_TICK
+					var/LZ
 					if(empty_turfs && empty_turfs.len)
-						var/LZ = empty_turfs[rand(empty_turfs.len-1)]
-						if (istype(beacon) && usingBeacon)
-							LZ = get_turf(beacon)
-							beacon.update_status(SP_LAUNCH)
-						SSshuttle.points -= SO.pack.cost
+						LZ = empty_turfs[rand(empty_turfs.len-1)]
+					else if (istype(beacon) && usingBeacon)

```

Comment:
This probably needs to be moved up above the `empty_turfs` check, as is a usable cargo bay area will prevent use of becons.

Reply:
goddamn im dumb thanks